src/Sylius/Bundle/ResourceBundle/Controller/RequestConfiguration.php

```diff
@@ -557,4 +557,28 @@ public function getGrid()
 
         return $this->parameters->get('grid');
     }
+
+    /**
+     * @return bool
+     */
+    public function hasStateMachine()
+    {
+        return $this->parameters->has('state_machine');
+    }
```

Comment:
The idea is to only have dedicated routes for applying these transitions, not generic update action, so this should fail if you try to apply invalid transition.